src/shared/dissect-image.c

```diff
@@ -337,6 +337,23 @@ static int timeout_handler(sd_event_source *s, uint64_t usec, void *userdata) {
         return r == -ENXIO ? -ETIMEDOUT : r;
 }
 
+static int retry_handler(sd_event_source *s, uint64_t usec, void *userdata) {
+        struct wait_data *w = userdata;
+        int r;
+
+        assert(w);
+
+        r = find_partition(w->parent_device, w->blkidp, w->timestamp_not_before, &w->found);
+        if (r != -ENXIO)
+                return r;
```

Comment:
ere too, shuldn't there be an sd_event_exit() on success?

Reply:
Ouch.